projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,365 @@
 # Entity Management
 
-**UNDER CONSTRUCTION**
+`withEntities` is an extension to facilitate CRUD operations for managing entities. It shares similarities with @ngrx/entity but is part of the SignalStore.
+
+It doesn't include methods for backend communication. You have to implement them separately.
+
+## Creating the Store
+
+A store implementing a `Todo` entity can have the following implementation:
+
+```typescript
+interface Todo {
+  id: number;
+  name: string;
+  finished: boolean;
+}
+
+const TodoStore = signalStore(withEntities<Todo>());
+```
+
+`withEntities` adds three properties of type `Signal` to the `TodoStore`.
+
+- `ids: Signal<EntityId[]>`: ids of all entities
+- `entities: Signal<Todo[]>`: array of all entities
+- `entityMap: Signal<EntityMap<Todo>>`> map of entities where the key is the id (`EntityId`)
+
+`entityMap` is a computed Signal that derives from the non-computed Signals `ids` and `entities`.
+
+By default, `withEntities` requires your entity to have a property of name `id`, which serves as a unique identifier. `id` has to be of type `EntityId`, which is a `number` or a `string`.
+
+## Adding & Setting
+
+`addEntity` and `addEntities` add entries to the store. They act as updater functions for `patchState`.
+
+Here is an example on how to use them inside a component.
+
+```typescript
+@Component({
+  selector: 'ngrx-todos',
+  template: `
+    <ul>
+      @for (todo of todoStore.entities(); track todo.id) {
+        <li>{{ todo.name }}</li>
+      }
+    </ul>
+  `,
+  standalone: true,
+  providers: [TodoStore],
+})
+export class TodosComponent implements OnInit {
+  todoStore = inject(TodoStore);
+
+  ngOnInit() {
+    // add a single entity
+    patchState(this.todoStore, addEntity({ id: 1, name: 'Car Washing', finished: false }));
+
+    // add multiple entities
+    patchState(
+      this.todoStore,
+      addEntities([
+        { id: 2, name: 'Car Washing', finished: false },
+        { id: 3, name: 'Room Cleaning', finished: false },
+      ])
+    );
+  }
+}
+```
+
+If you add an entity with an existing id, the original entity is not overwritten, and no error is thrown.
+
+In this example, the todo remains as "Cat Feeding".
+
+```typescript
+patchState(this.todoStore, addEntity({ id: 1, name: 'Cat Feeding', finished: false }));
+
+patchState(this.todoStore, addEntity({ id: 1, name: 'Dog Feeding', finished: false }));
+```
+
+---
+
+Updaters `setEntity` and `setEntities` are used to add new or replace existing entities from a collection.
+
+In this example, we add a new `Todo` entity with "Cat Feeding" and replace it with "Dog Feeding".
+
+```typescript
+patchState(this.todoStore, setEntity({id: 1, name: 'Cat Feeding', finished: false}));
+
+patchState(this.todoStore, setEntity({id: 1, name: 'Dog Feeding', finished: false}));
+
+// version with setEntities and two different entities
+
+patchState(this.todoStore, setAllEntities([
```

Comment:
It seems that `setEntities` is accidentally replaced with `setAllEntities` in this example

Reply:
Done